Automattic/woocommerce-payments
# Import fingerprint package directly instead of lazy loading

Fixes #5224 

#### Changes proposed in this Pull Request





- Remove lazy loading as it may be causing compatibility issues with script concatenators plugins.
- Improve the error message in case anything fails.





#### Testing instructions





* Unit tests should be passing
* Apply the following patch
```diff
diff --git a/client/checkout/utils/fingerprint.js b/client/checkout/utils/fingerprint.js
index 101a177d..354aa604 100644
--- a/client/checkout/utils/fingerprint.js
+++ b/client/checkout/utils/fingerprint.js
@@ -10,9 +10,9 @@ export const FINGERPRINT_GENERIC_ERROR =
 export const getFingerprint = async () => {
 	const agent = await FingerprintJS.load( { monitoring: false } );
 
-	if ( ! agent ) {
+	// if ( ! agent ) {
 		throw new Error( FINGERPRINT_GENERIC_ERROR );
-	}
+	// }
 
 	return await agent.get();
 };
```
* Enable UPE in **WCPay Dev**
* Go to your shop, add a product to your cart and open the checkout page
* It should show the `An error was encountered when preparing the payment form. Please try again later.` error message
* Also test the same flow for
  * UPE checkout - Block checkout
  * Non-UPE checkout - Classic checkout

-------------------


**Post merge**
